contracts/permissionless/OrderList.sol

```diff
@@ -281,7 +281,7 @@ contract OrderList is PermissionGroups, OrderListInterface {
         Order storage order = orders[orderId];
 
         // Make sure such order exists in mapping.
-        require(order.prevId != 0 || order.nextId != 0);
+        require(order.prevId != 0 && order.nextId != 0 && order.prevId != TAIL_ID && order.nextId != HEAD_ID);
```

Comment:
Please also add tests to `orderlist.js` to check these different requirements directly.

Reply:
will fix documentation here.
but not sure I can follow your strict conventions.